invenio_records_resources/services/records/queryparser/transformer.py

```diff
@@ -56,11 +98,19 @@ def visit_search_field(self, node, context):
         # Use the node name if not mapped for transformation.
         term_name = self._mapping.get(node.name, node.name)
         field_value_mapper = None
-
         if isinstance(term_name, FieldValueMapper):
             field_value_mapper = term_name
             term_name = field_value_mapper.term_name
-
+        if isinstance(term_name, RestrictedTerm):
+            field_value_mapper = term_name
+            allows = term_name.allows(context["identity"])
+            term_name = node.name
+            if not allows:
+                raise QuerystringValidationError(
+                    _("Invalid search field: {field_name}.").format(
+                        field_name=node.name
+                    )
+                )
```

Comment:
I would move the assignments afterwards, in the success branch, to minimise noise
```suggestion
            if not  term_name.allows(context["identity"]):
                raise QuerystringValidationError(
                    _("Invalid search field: {field_name}.").format(
                        field_name=node.name
                    )
                )
                
            field_value_mapper = term_name
            term_name = node.name
```